harness/determined/constants.py

```diff
@@ -24,16 +26,39 @@
     "optimizations": DEFAULT_OPTIMIZATIONS,
 }
 
+# Allow users to override the ports using the environment variables, in case
+# there are other processes running on the compute node which are already using
+# our default port. Users will likely set these in their "startup-hook.sh".
+# Below is an example how to set these in the experiment's YAML file for testing.
```

Comment:
I spoke about this strategy for a while with @stoksc, and I agree that the overall approach is the right approach for now.  But I don't really want to give this feature "first-class" status, because some day a port registry or some sort of choose-port-then-allgather-chosen-port-numbers logic is probably a better strategy.

My real concern is that there would be people who would see this port-configuration logic you have here and would coopt it to accomplish something unrelated to the collision we are currently trying to avoid, and when we have a port registry or something, then we will have trouble removing these environment variables.

I think the right approach is to just set user expectations in this comment.  This isn't a first-class feature and we aren't promising it's always around.  So instead of talking about it like a normal feature, like "allow users to override... users will likely...", let's use phrasing more like "expose a temporary workaround in the rare case that ... until we implement a more automatic solution".

Reply:
Will change the comment.